src/status_im2/contexts/chat/messages/pin/banner/view.cljs

```diff
@@ -3,19 +3,37 @@
             [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
+(defn resolve-message [parsed-text]
```

Comment:
@rasom, I think this kind of code should live in a subscription, since it doesn't have Hiccup. All the data the function needs is available from the `parsed-text` arg and the app db, so it fits like a glove in a new subscription. A subscription unit test for this recursive function would be good too.

Reply:
@ilmotta Just to clarify details of that new subscription: do you mean a subscription like `[:chats/last-pinned-message-text chat-id]` which returns resolved message text or just a separate sub to get a raw `parsed-text`?

The issue with a former one is that we don't actually know how many `subs` that `parsed-text` is going to require as well as we don't know parameters of those subs. So we basically would need to extract `parsed-text` before we can create necessary `:messages/resolve-mention` subs which kind of breaks `reg-sub` because doing that in a signal fn is not cool (in case if that message will be edited and mentions will be changed we are not going to get them as we subscribed to old ones), and doing that in a handler fn is also not cool as it should be a pure fn. Doing a full derivation of mentions from `app-db` is again not cool, missing the entire point of extractors and possibly creating unnecessary coupling with contacts/mentions subs code. 
Well, so, not sure I like it. But also maybe I'm missing some technic to do it here without getting into another trouble.

The latter option is fine though then `resolve-message` is not going to make it to subs, but it is pretty testable without that, if that's a point.

So sir, tell me please, what do I have to do?

Reply:
It seems my assumption was incorrect. I assumed `parsed-text` *could* come as a single piece of data from the app db, so deriving the *resolved message* in a new sub would be a natural idea, to me at least. My thinking was, if `parsed-text` exists in the app db, and `resolve-message` only does pure processing and relies on the subscription `:messages/resolve-mention`, then a subscription seems appropriate instead of processing at the view layer (to me at least). But if this means doing "full derivation of mentions from app-db", then as you said, of course it's not cool.

You have a better understanding about this code than I do, so if you think it's better to keep it as is, then it's just fine ;)

Worth mentioning, the way `resolve-message` is implemented right now is not optimal already IMO, since we're using reactions inside a recursive function. It works, but reactions have considerable overhead, and I like to think we should keep them out of loops if possible. But you know, this is fine, I mean, we do this in a lot of places. I've seen one or two mobile PRs where this whole "don't subscribe inside loops" idea was brought up, so maybe something for us to consider and measure/compare.

> The latter option is fine though then resolve-message is not going to make it to subs, but it is pretty testable without that, if that's a point.

You're right, the function is totally testable as is, though it lives in the view layer and doesn't produce hiccup. It would be nice if you could test it :)

> So sir, tell me please, what do I have to do?

No no sir, you're well equipped to judge what's best :) I'd say keep it as is if you think a new sub is not the way to go.

Thanks for sharing your thoughts in such detail.

Reply:
@ilmotta got it, thank!

added a small test meanwhile